smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -70,17 +74,33 @@ public ApplicationProtocol getApplicationProtocol() {
         return ApplicationProtocol.createDefaultHttpApplicationProtocol();
     }
 
+    @Override
+    public void generateSharedComponents(GenerationContext context) {
+        TypeScriptWriter writer = context.getWriter();
+
+        SymbolReference responseType = getApplicationProtocol().getResponseType();
+        writer.addImport("ResponseMetadata", "ResponseMetadata", "@aws-sdk/types");
```

Comment:
You may want to alias this as "__ResponseMetadata" on the off-chance that someone defines this shape in their model.